src/main/java/impl/image/ConvolutionProviderFactory.java

```diff
@@ -2,8 +2,41 @@
 
 import api.image.ConvolutionProvider;
 
+import java.awt.*;
+import java.lang.Math.*;
 public class ConvolutionProviderFactory {
-    public static ConvolutionProvider getInstance() {
-        return null;
+    private static ConvolutionProviderFactory instance;
+    public static ConvolutionProviderFactory getInstance()
+    {
+        if (instance == null)instance = new ConvolutionProviderFactory();
+        return instance;
     }
+    public Color[][] apply(Color[][] image, double[][] kernel){
+        int n=image.length,m=image[0].length,kn =kernel.length,km=kernel[0].length;
```

Comment:
You should initialize one variable per line.
https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141270.html#2992